optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/GenericJaxbIO.java

```diff
@@ -184,6 +184,14 @@ public T readAndValidate(Document document, Schema schema) {
         }
     }
 
+    private String validationEventToString(ValidationEvent validationEvent) {
+        return new StringBuilder()
+                .append(validationEvent.getMessage())
+                .append("\n")
+                .append("Node: ").append(validationEvent.getLocator().getNode().getNodeName())
+                .toString();
```

Comment:
Use string concatenation, as that's equal or faster than a StringBuilder if no loops are invovled (java compiler actually translates it into a StringBuilder but deals better with capacity iirc)
Also reads better:

return validationEvent.getMessage() + "n" + "Node: " + validationEvent.getLocator().getNode().getNodeName()

=> at which point I'd inline it